app/services/padlock/token_generator.rb

```diff
@@ -7,15 +7,21 @@ def initialize(params = {})
     end
 
     def call
-      user.tokens.create(
-        key:        SecureRandom.base64(18),
-        key_type:   type,
-        expired_at: expired_at
-      )
+      ExpireTokenJob.set(wait_until: token.expired_at).perform_now(token_id: token.id) if token
```

Comment:
This seems to work, but, and I know a good but when I see one, that's probably not how it's supposed to be used. Please advise